commit 16096ae93a26b4df98e646be4e27a93e2126f0aa
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Sat Mar 3 14:18:03 2018 +0000

    overlays: Add 'upstream' overlay
    
    The 'upstream' overlay makes the necessary changes for a downstream
    .dtb to be used with an upstream kernel. It is currently made up from
    three other overlays - vc4-kms-v3d, dwc2 and upstream-aux-interrupt.
    The VPU firmware will soon be made to automatically load this overlay
    when an upstream kernel is detected (using the trailer supplied by the
    mkknlimg script).
    
    See: https://github.com/raspberrypi/linux/pull/2393
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/arch/arm/boot/dts/overlays/Makefile b/arch/arm/boot/dts/overlays/Makefile
index e3a7218de94a..64ea686d6495 100644
--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -123,6 +123,7 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	tinylcd35.dtbo \
 	uart0.dtbo \
 	uart1.dtbo \
+	upstream.dtbo \
 	upstream-aux-interrupt.dtbo \
 	vc4-fkms-v3d.dtbo \
 	vc4-kms-v3d.dtbo \
diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index 3fb55032aa83..6db162901e3a 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -1742,9 +1742,17 @@ Params: txd1_pin                GPIO pin for TXD1 (14, 32 or 40 - default 14)
         rxd1_pin                GPIO pin for RXD1 (15, 33 or 41 - default 15)
 
 
+Name:   upstream
+Info:   Allow usage of downstream .dtb with upstream kernel. Comprises
+        vc4-kms-v3d, dwc2 and upstream-aux-interrupt overlays.
+Load:   dtoverlay=upstream
+Params: <None>
+
+
 Name:   upstream-aux-interrupt
 Info:   Allow usage of downstream .dtb with upstream kernel by binding AUX
-        devices directly to the shared AUX interrupt line.
+        devices directly to the shared AUX interrupt line. One of the parts
+        of the 'upstream' overlay
 Load:   dtoverlay=upstream-aux-interrupt
 Params: <None>
 
diff --git a/arch/arm/boot/dts/overlays/upstream-overlay.dts b/arch/arm/boot/dts/overlays/upstream-overlay.dts
new file mode 100644
index 000000000000..3890d900486f
--- /dev/null
+++ b/arch/arm/boot/dts/overlays/upstream-overlay.dts
@@ -0,0 +1,154 @@
+// redo: ovmerge -c vc4-kms-v3d-overlay.dts,cma-96 dwc2-overlay.dts,dr_mode=otg upstream-aux-interrupt-overlay.dts,
+
+/dts-v1/;
+/plugin/;
+
+#include <dt-bindings/clock/bcm2835.h>
+
+/ {
+	compatible = "brcm,bcm2835", "brcm,bcm2708", "brcm,bcm2709";
+	fragment@0 {
+		target-path = "/chosen";
+		__dormant__ {
+			bootargs = "cma=256M";
+		};
+	};
+	fragment@1 {
+		target-path = "/chosen";
+		__dormant__ {
+			bootargs = "cma=192M";
+		};
+	};
+	fragment@2 {
+		target-path = "/chosen";
+		__dormant__ {
+			bootargs = "cma=128M";
+		};
+	};
+	fragment@3 {
+		target-path = "/chosen";
+		__overlay__ {
+			bootargs = "cma=96M";
+		};
+	};
+	fragment@4 {
+		target-path = "/chosen";
+		__dormant__ {
+			bootargs = "cma=64M";
+		};
+	};
+	fragment@5 {
+		target = <&i2c2>;
+		__overlay__ {
+			status = "okay";
+		};
+	};
+	fragment@6 {
+		target = <&fb>;
+		__overlay__ {
+			status = "disabled";
+		};
+	};
+	fragment@7 {
+		target = <&pixelvalve0>;
+		__overlay__ {
+			interrupts = <2 13>;
+			status = "okay";
+		};
+	};
+	fragment@8 {
+		target = <&pixelvalve1>;
+		__overlay__ {
+			interrupts = <2 14>;
+			status = "okay";
+		};
+	};
+	fragment@9 {
+		target = <&pixelvalve2>;
+		__overlay__ {
+			interrupts = <2 10>;
+			status = "okay";
+		};
+	};
+	fragment@10 {
+		target = <&hvs>;
+		__overlay__ {
+			interrupts = <2 1>;
+			status = "okay";
+		};
+	};
+	fragment@11 {
+		target = <&hdmi>;
+		__overlay__ {
+			interrupts = <2 8>, <2 9>;
+			status = "okay";
+		};
+	};
+	fragment@12 {
+		target = <&v3d>;
+		__overlay__ {
+			interrupts = <1 10>;
+			status = "okay";
+		};
+	};
+	fragment@13 {
+		target = <&vc4>;
+		__overlay__ {
+			status = "okay";
+		};
+	};
+	fragment@14 {
+		target-path = "/soc/dma";
+		__overlay__ {
+			brcm,dma-channel-mask = <0x7f35>;
+		};
+	};
+	fragment@15 {
+		target = <&clocks>;
+		__overlay__ {
+			claim-clocks = <BCM2835_PLLD_DSI0 BCM2835_PLLD_DSI1 BCM2835_PLLH_AUX BCM2835_PLLH_PIX>;
+		};
+	};
+	fragment@16 {
+		target = <&vec>;
+		__overlay__ {
+			status = "okay";
+		};
+	};
+	fragment@17 {
+		target = <&usb>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		dwc2_usb: __overlay__ {
+			compatible = "brcm,bcm2835-usb";
+			reg = <0x7e980000 0x10000>;
+			interrupts = <1 9>;
+			dr_mode = "otg";
+			g-np-tx-fifo-size = <32>;
+			g-rx-fifo-size = <256>;
+			g-tx-fifo-size = <512 512 512 512 512 768>;
+			status = "okay";
+		};
+	};
+	fragment@18 {
+		target = <&uart1>;
+		__overlay__ {
+			interrupt-parent = <&intc>;
+			interrupts = <0x1 0x1d>;
+		};
+	};
+	fragment@19 {
+		target = <&spi1>;
+		__overlay__ {
+			interrupt-parent = <&intc>;
+			interrupts = <0x1 0x1d>;
+		};
+	};
+	fragment@20 {
+		target = <&spi2>;
+		__overlay__ {
+			interrupt-parent = <&intc>;
+			interrupts = <0x1 0x1d>;
+		};
+	};
+};
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1203_16096ae93a26b4df98e646be4e27a93e2126f0aa.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1203_16096ae93a26b4df98e646be4e27a93e2126f0aa.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
